adaptive_memory_v3.1.py

```
@@ -4303,11 +4544,129 @@ async def _get_embedding(self, text: str) -> Optional[np.array]:
+        """
+        try:
+            # Use data directory for persistence
+            cache_dir = "/app/backend/data/cache/embeddings"
+            os.makedirs(cache_dir, exist_ok=True)
+            cache_file = os.path.join(cache_dir, f"{user_id}_embeddings.json")
+            
+            # Load existing cache
+            cache = {}
+            if os.path.exists(cache_file):
+                try:
+                    with open(cache_file, 'r') as f:
```

Ruff (0.14.10)

Async functions should not open files with blocking methods like `open`

(ASYNC230)